### PR TITLE
boards/nucleo-f446ze: add support for ADC

### PR DESCRIPTION
**Pull request by krzysztof-cabaj**
### Contribution description

This PR adds to the Nucleo-f446ze ADC configuration.

### Testing procedure

Flash the board using `tests/periph_adc` program. Check if measured values changes when A0-A5 pins are
connected to the 3,3V, GND or to the potentiometer. 

### Issues/PRs references

None

### CODE REVIEW

File: boards/nucleo-f446ze/Makefile.features [hunk 1]
```diff
@@ -8,6 +8,7 @@ FEATURES_PROVIDED += periph_can
 FEATURES_PROVIDED += periph_pwm
 FEATURES_PROVIDED += periph_rtc
 FEATURES_PROVIDED += periph_spi
+FEATURES_PROVIDED += periph_adc
```
Comment (benpicco) [inline boards/nucleo-f446ze/Makefile.features:11]: Kconfig needs an update too

Comment (krzysztof-cabaj) [reply boards/nucleo-f446ze/Makefile.features:11]: I'm not familiar with the Kconfig - but i hope one line from [commit](https://github.com/RIOT-OS/RIOT/pull/18647/commits/f13bc78fab055309dacaab540569b815864ae7fd) is fine.

Comment (benpicco) [reply boards/nucleo-f446ze/Makefile.features:11]: Yes that will work - please squash

File: boards/nucleo-f446ze/include/periph_conf.h [hunk 2]
```diff
@@ -169,6 +169,45 @@ static const spi_conf_t spi_config[] = {
 #define SPI_NUMOF           ARRAY_SIZE(spi_config)
 /** @} */
 
+
+/**
+ * @name    ADC configuration
```
Comment (benpicco) [inline boards/nucleo-f446ze/include/periph_conf.h:174]:
```suggestion
 * @brief    ADC configuration
```

File: boards/nucleo-f446ze/include/periph_conf.h [hunk 3]
```diff
@@ -169,6 +169,45 @@ static const spi_conf_t spi_config[] = {
 #define SPI_NUMOF           ARRAY_SIZE(spi_config)
 /** @} */
 
+
```
Comment (benpicco) [inline boards/nucleo-f446ze/include/periph_conf.h:172]:
```suggestion
```

File: boards/nucleo-f446ze/include/periph_conf.h [hunk 4]
```diff
@@ -169,6 +169,45 @@ static const spi_conf_t spi_config[] = {
...
+};
+
+/**
+ * @brief Number of ADC devices
+ */
+#define ADC_NUMOF           ARRAY_SIZE(adc_config)
+
+/** @} */
+
+
```
Comment (benpicco) [inline boards/nucleo-f446ze/include/periph_conf.h:210]:
```suggestion
```

**Comment by krzysztof-cabaj** [comment]
@benpicco I added editorial fixes and squashed all commits.

**Review by benpicco — APPROVED** [review]

File: boards/nucleo-f446ze/include/periph_conf.h [hunk 5]
```diff
@@ -169,6 +169,43 @@ static const spi_conf_t spi_config[] = {
...
+ * Table 18, page 56.
+ * @{
+ */
+static const adc_conf_t adc_config[] = {
+    { .pin = GPIO_PIN(PORT_A, 0), .dev = 2, .chan =  3 }, /* ADC123_IN3   */
+    { .pin = GPIO_PIN(PORT_A, 1), .dev = 2, .chan = 10 }, /* ADC123_IN10  */
+    { .pin = GPIO_PIN(PORT_A, 4), .dev = 2, .chan = 13 }, /* ADC123_IN13  */
+    { .pin = GPIO_PIN(PORT_B, 0), .dev = 2, .chan =  9 }, /* ADC123_IN9   */
+    { .pin = GPIO_PIN(PORT_C, 1), .dev = 2, .chan = 15 }, /* ADC3_IN15    */
+    { .pin = GPIO_PIN(PORT_C, 0), .dev = 2, .chan =  8 }, /* ADC3_IN8     */
```
Comment (benpicco) [inline boards/nucleo-f446ze/include/periph_conf.h:199]: You can also add

```C
    { .pin = GPIO_UNDEF, .dev = 0, .chan =  18 }, /* VBAT     */
```

and then 

```
#define VBAT_ADC            ADC_LINE(6) /**< VBAT ADC line */
```

Then `tests/periph_vbat` should work / pass CI

**Comment by krzysztof-cabaj** [comment]
@benpicco I added VBAT and now MurdockCI i green. 

Thank you for help.

**Pull request merged**